### DISCUSSION TITLE
How to access the hierarchy and then Three.js changes each object.

### DISCUSSION THREAD

**Discussion by bigbea1r** [google/model-viewer#4712] [Show and tell]
I have a task. Change the model imported into modelviewer and editor it in Three.js. How to do it? I need this in order to make an editor based on modelviewer. I read all the documentation, but found only a few items:
 You are also welcome to hack/fork us. You can use mvElement[Object.getOwnPropertySymbols(mvElement).find(e => e.description === 'scene')]; to get access to our internal three.js scene and do whatever you want. And what you link to [Post Process Effects](https://modelviewer.dev/examples/postprocessing)
 
 The code currently looks like this:
 
 <!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta name="description" content="<model-viewer> scene graph examples">
    <title>Editor Table</title>
    <link rel="stylesheet" href="main.css">
    <script type="module" src="https://ajax.googleapis.com/ajax/libs/model-viewer/3.4.0/model-viewer.min.js"></script>
</head>
<body>
    
    <model-viewer id="tableModel" camera-controls touch-action="pan-y" 
    src="table.glb" tone-mapping="commerce" 
    ar alt="A 3D model of a Table">
        <div class="progress-bar hide" slot="progress-bar">
            <div class="update-bar"></div>
        </div>
        <button slot="ar-button" id="ar-button">
            View in your space
        </button>
        <div id="ar-prompt">
            <img src="https://modelviewer.dev/shared-assets/icons/hand.png">
        </div>
        <div class="controls">
            Variant: <select id="variant"></select>
        </div>
    </model-viewer>
    <div class="interface">
        <button id="textures">Текстуры</button>
        <button id="sizes">Высота</button>
        <button id="state">Размер</button>
    </div>
    <script>
        const modelViewer = document.querySelector("model-viewer#tableModel");
        modelViewer.model.hierarchy[1].name = 'hack'
        console.log(modelViewer);
    </script>
</body>
</html>

**Comment by elalish** [comment]
What kind of edits do you want to do? If it's just materials, you can use our materials API: https://modelviewer.dev/examples/scenegraph/#pickMaterialExample

Where did `modelViewer.model.hierarchy[1]` come from? If you want to hack MV, you'll have to read our code - we can't support you here.
